src/plugin.js

```diff
@@ -72,6 +83,86 @@
     });
   });
 
+
+
+  // Backfill support for %-based dimensions in players. Includes copy/
+  // pasted code from the `updateStyleEl_` method in video.js 5 because
+  // that hard-codes px. :-/
+  Player.prototype.dimension = function(dimension, value) {
+
+    // When the value is set as a %, bypass the usual handling.
+    if (isPct(value)) {
+      this[dimension + '_'] = value;
+      this.updateStyleEl_();
```

Comment:
Because the original `updateStyleEl_` has `px` hard-coded:

```js
    stylesheet.setTextContent(this.styleEl_, `
      .${idClass} {
        width: ${width}px;
        height: ${height}px;
      }

      .${idClass}.vjs-fluid {
        padding-top: ${ratioMultiplier * 100}%;
      }
    `);
```